libraries/chain/host_context.cpp

```diff
@@ -39,54 +39,83 @@ host_context::~host_context() = default;
 
 // called from apply_context or sync_call_context
 int64_t host_context::execute_sync_call(name call_receiver, uint64_t flags, std::span<const char> data) {
-   const uint32_t depth = sync_call_depth + 1;
-   const auto max_depth = control.get_global_properties().configuration.max_sync_call_depth;
-   EOS_ASSERT(depth <= max_depth, sync_call_depth_exception,
-              "reached sync call max call depth ${max_depth}", ("max_depth", max_depth));
+   auto start = fc::time_point::now();
 
-   const auto max_data_size = control.get_global_properties().configuration.max_sync_call_data_size;
-   EOS_ASSERT(data.size() <= max_data_size, sync_call_call_data_exception,
-              "sync call call data size must be less or equal to ${max_data_size} bytes", ("max_data_size", max_data_size));
+   // As early as possible, create the call trace of this new sync call in the parent's
+   // (sender's) trace to record entire trace of the sync call, including any exceptions
+   auto& trace = get_root_action_trace();
+   const bool read_only = flags & static_cast<uint64_t>(sync_call_flags::read_only);
```

Comment:
In an earlier PR we discussed using `has_field()` not sure how much it matters

Reply:
Because the last element `all_allowed_bits` in the enum `sync_call_flags` is not a flag (or field in a sense). I did not use the `has_field` mechanism. Will revisit this later.